### PR TITLE
Automate parameters handler upload

### PR DESCRIPTION
**Pull request by johnfolly**
# Automate parameters handler upload

## Options:
  `-h`, `--help`  Show help message and exit.

## Commands:
  `download`  Download a solution handler zip.
  `upload`    Upload a solution handler zip.

#### Usage:  `babylon api solution handler download [OPTIONS] `

  Download a solution handler zip.

  Requires key `solution_id` in the deployment config file.

  Requires key `organization_id` in the deployment config file.

#### Options:
  -r, --run-template TEXT         The run Template identifier  [required]
  -t, --handler-type [parameters_handler|validator|prerun|engine|postrun|scenariodata_transform] Handler type  [required]
  -h, --help                               Show help message and exit.

#### Usage:   `babylon api solution handler upload [OPTIONS] HANDLER_PATH`

Upload a solution handler zip.

  Requires key `solution_id` in the deployment config file.

  Requires key `organization_id` in the deployment config file.

#### Options:
  -r, --run-template TEXT         The run Template identifier  [required]
  -t, --handler-type [parameters_handler|validator|prerun|engine|postrun|scenariodata_transform] Handler type  [required]
  -h, --help                               Show help message and exit.

### CODE REVIEW

File: Babylon/groups/api/groups/solution/groups/handler/commands/download.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    except ServiceException:
+        logger.error(f"Organization with id {organization_id} and or Solution {solution_id} not found.")
+        return
+
+    if not response:
+        logger.error("No retrieved handler")
+        return
+
+    with open(run_template_id + ".zip", "w") as _f:
+        IO.write(response, _f)
```
Comment (o-reo) [inline Babylon/groups/api/groups/solution/groups/handler/commands/download.py:86]: Usually we use
```suggestion
        _f.write(response)
```

**Review by o-reo — COMMENTED** [review]
Just a minor change suggested, thanks :+1:

File: Babylon/groups/api/groups/solution/groups/handler/commands/download.py (hunk 1 above)
Comment (johnfolly) [reply Babylon/groups/api/groups/solution/groups/handler/commands/download.py:86]: the response is an zip file and if use _f.write(response) i got  this erro message

```
│   85 │   with open(run_template_id + ".zip", "w") as _f:                                         │
│ ❱ 86 │   │   _f.write(response)                                                                  │
│   87 │                                                                                           │
│   88 │   logger.info(f"{handler_id} handler downloaded from solution {solution_id} successful    │
│   89                                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: write() argument must be str, not _io.BufferedReader
```

Comment (o-reo) [reply Babylon/groups/api/groups/solution/groups/handler/commands/download.py:86]:
```python
    with open(run_template_id + ".zip", "wb") as _f:
        _f.write(response, _f)
```

Comment (johnfolly) [reply Babylon/groups/api/groups/solution/groups/handler/commands/download.py:86]:
```
│   89 │   with open(f"{run_template_id}.zip", "wb") as _f:                                        │
│ ❱ 90 │   │   _f.write(response)                                                                  │
│   91 │                                                                                           │
│   92 │   logger.debug(response)                                                                  │
│   93 │   logger.info(                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: a bytes-like object is required, not '_io.BufferedReader'
```

Type error

Comment (o-reo) [reply Babylon/groups/api/groups/solution/groups/handler/commands/download.py:86]: [Maybe this?](https://bobbyhadz.com/blog/python-typeerror-a-bytes-like-object-is-required-not-io-bytesio)
```suggestion
        _.write(response.read())
```

**Review by o-reo — APPROVED** [review]
:rocket:

**Pull request merged**